src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenAsyncMainTests.cs

```diff
@@ -2063,5 +2063,397 @@ static async MyTask Main()
+                  IL_000b:  ldloca.s   V_0
+                  IL_000d:  call       "void System.Runtime.CompilerServices.TaskAwaiter.GetResult()"
+                  IL_0012:  ret
+                }
+                """);
+        }
+
+        [Fact]
+        public void AsyncMainFallbackToOldPattern_RedefinedTaskT()
+        {
+            var source = """
+                #pragma warning disable CS0436
+
+                class Program
+                {
+                    static async System.Threading.Tasks.Task<int> Main()
+                    {
+                        throw null;
+                    }
+                }
+
+                namespace System.Threading.Tasks
+                {
+                    public class Task<TResult>
+                    {
+                        public System.Runtime.CompilerServices.TaskAwaiter<TResult> GetAwaiter() => throw null;
+                    }
+                }
+
+                namespace System.Runtime.CompilerServices
+                {
+                    public struct AsyncTaskMethodBuilder<TResult>
+                    {
+                        public System.Threading.Tasks.Task<TResult> Task => null;
+                        public static AsyncTaskMethodBuilder<TResult> Create() => throw null;
+                        public void SetException(Exception exception){}
+                        public void SetResult(TResult result){}
+                        public void AwaitOnCompleted<TAwaiter,TStateMachine>(ref TAwaiter awaiter, ref TStateMachine stateMachine) where TAwaiter : System.Runtime.CompilerServices.INotifyCompletion where TStateMachine : System.Runtime.CompilerServices.IAsyncStateMachine{}
+                        public void AwaitUnsafeOnCompleted<TAwaiter,TStateMachine>(ref TAwaiter awaiter, ref TStateMachine stateMachine) where TAwaiter : System.Runtime.CompilerServices.ICriticalNotifyCompletion where TStateMachine : System.Runtime.CompilerServices.IAsyncStateMachine{}
+                        public void Start<TStateMachine>(ref TStateMachine stateMachine) where TStateMachine : System.Runtime.CompilerServices.IAsyncStateMachine{}
+                        public void SetStateMachine(System.Runtime.CompilerServices.IAsyncStateMachine stateMachine){}
+                    }
+
+                    public static class AsyncHelpers
+                    {
+                        public static int HandleAsyncEntryPoint(Threading.Tasks.Task<int> task)
+                        {
+                            return task.GetAwaiter().GetResult();
+                        }
+                    }
+                }
+                """;
+
+            var corlibRef = CreateEmptyCompilation(MinimalAsyncCorelibWithAsyncHelpers, assemblyName: "mincorlib").EmitToImageReference();
+            var comp = CreateEmptyCompilation(source, references: new[] { corlibRef }, options: TestOptions.DebugExe);
+
+            var verifier = CompileAndVerify(comp, verify: Verification.Skipped);
```

Comment:
>CompileAndVerify [](http://example.com/codeflow?start=27&length=16)

It looks like none of the tests verify runtime behavior with new codegen. Let's add skipped test(s) that do that and open an issue to enable the tests once we get a runtime supporting the API.